nasa-gibs/worldview
# wv-2836 satellite names

## Description
We would like to add the satellite name next to the orbit tracks in the layer picker. This is an extension of wv-2702, which added satellite names for associated layers in the sidebar. 

Fixes # wv-2836

## How To Test
- git checkout wv-2836-satellite-names
- npm ci
- npm run watch

While this targets the standard layer picker, there are a number of places that we want to confirm the orbit track labels are correct.

**Testing orbit track strings in the standard layer picker:**
- Click Add Layers
- Navigate to Featured tab
- Confirm under "Orbital Tracks" the labels are correct "(Ascending/Descending)/(Day/Night) (Satellite Name)"

**Testing orbit track strings in the sidebar:**
- Load a layer with associated orbital layers (ex. Opera Dynamic Surface Water Extent L3)
- Click "View Options"
- Confirm the Associated layers text is proper "Orbit Track - (Ascending/Descending)/(Day/Night) (Satellite Name)"
- Enable at least one orbit track for this layer
- In the sidebar, confirm satellite name (next to color swatch) is correct "(Ascending/Descending)/(Day/Night) (Satellite Name)"
- In the sidebar, hover over the palette swatch for an orbit track. Confirm tooltip is correct "Acquisition time (UTC) (Ascending/Descending)/(Day/Night) (Satellite Name)"

**Testing orbit track strings in the Recent tab of the layer picker:**
- Click Add Layers
- Go to Recent tab
- Clear list
- Go to Features tab
- Select any orbit track
- Go back to Recent tab
- Confirm Orbit Track layer labels are correct "(Satellite Name) - Orbit Track & Time(Ascending/Descending)/(Day/Night)"

**Testing granule associated layers (wv-2840 on the board for more details):**

- Add Granule Corrected reflectance layer
- Select View Options in the sidebar
- Confirm that the associated layer names are correct (no duplication of labels)

## Merging

Please use the `squash and merge` commit method unless each commit in your branch is vital to the commit history of main.

@nasa-gibs/worldview
